openedx/core/djangoapps/course_groups/views.py

```diff
@@ -379,3 +408,41 @@ def debug_cohort_mgmt(request, course_key_string):
         kwargs={'course_key': text_type(course_key)}
     )}
     return render_to_response('/course_groups/debug.html', context)
+
+
+@view_auth_classes()
+@expect_json
+def api_cohort_settings(request, course_key_string):
+    """
+    OAuth2 endpoint for cohort settings.
+    """
+    return _course_cohort_settings_handler(request, course_key_string)
+
+
+@view_auth_classes()
+@expect_json
+def api_cohort_handler(request, course_key_string, cohort_id=None):
+    """
+    OAuth2 endpoint for cohort handler.
+    """
+    if request.method == 'POST':
+        get_course_with_access(request.user, 'staff', CourseKey.from_string(course_key_string))
+        return api.add_users_to_cohorts(request, course_key_string)
+    return _cohort_handler(request, course_key_string, cohort_id)
+
+
+@view_auth_classes()
+def api_cohort_users(request, course_key_string, cohort_id, username=None):
```

Comment:
I think it would be better to make this a class-based view.

Reply:
@xitij2000, @viadanna, DRF Viewsets can be used to write the CRUD logic in a single class.

Reply:
Certainly! That is a better idea and more appropriate here.